thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/ThunderbirdMboxFileIngestModule.java

```diff
@@ -435,38 +436,39 @@ static String getRelModuleOutputPath() throws NoCurrentCaseException {
      * @param abstractFile
      */
     private void processEmails(List<EmailMessage> partialEmailsForThreading, Iterator<EmailMessage> fullMessageIterator, AbstractFile abstractFile) {
-        
+
         // Putting try/catch around this to catch any exception and still allow
         // the creation of the artifacts to continue.
-        try{
+        try {
             EmailMessageThreader.threadMessages(partialEmailsForThreading);
-        } catch(Exception ex) {
+        } catch (Exception ex) {
             logger.log(Level.WARNING, String.format("Exception thrown parsing emails from %s", abstractFile.getName()), ex);
         }
-        
+
         List<AbstractFile> derivedFiles = new ArrayList<>();
 
         int msgCnt = 0;
-        while(fullMessageIterator.hasNext()) {
+        while (fullMessageIterator.hasNext()) {
             EmailMessage current = fullMessageIterator.next();
-            
-            if(current == null) {
+
+            if (current == null) {
                 continue;
             }
 
-            if(partialEmailsForThreading.size() > msgCnt) {
+            if (partialEmailsForThreading.size() > msgCnt) {
                 EmailMessage threaded = partialEmailsForThreading.get(msgCnt++);
-                
-                if(threaded.getMessageID().equals(current.getMessageID()) &&
-                        threaded.getSubject().equals(current.getSubject())) {
-                    current.setMessageThreadID(threaded.getMessageThreadID());
-                }
+
+                if ((threaded.getMessageID() != null) &&
```

Comment:
It's fine to add the null check, but the way EmailMessage is suppose to work is that all the functions that return strings should return empty string instead of null.  So that means somewhere in the threading code the the message id is being set to null instead of empty string.

Reply:
I took out the code I previously put in and made a change to setMessageID to check for null and return "" if it was.